splice_machine/src/main/java/com/splicemachine/derby/stream/function/NLJoinFunction.java

```diff
@@ -125,8 +125,10 @@ public void close() {
                 LOG.error("Exception while closing operation", e);
             }
         }
-        for (Future<Pair<OperationContext, Iterator<ExecRow>>> future : futures) {
-            future.cancel(true);
+        if (futures != null) {
```

Comment:
@dgomezferro Could you take a look at my comment at https://splicemachine.atlassian.net/browse/DB-7126?focusedCommentId=54759&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-54759 and see if it is a valid concern with the fix to just check futures !=null.

Reply:
@yxia92 I've answered on that JIRA with more details, but I don't think that's problematic for this case. There might be some issues with CachedOperations below NLJoins or other operations that clone() the operation tree though.